hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/model/EntityInterceptorListenerTest.java

```diff
@@ -0,0 +1,188 @@
+import org.eclipse.hawkbit.repository.model.Target;
+import org.junit.Test;
+
+import ru.yandex.qatools.allure.annotations.Description;
+import ru.yandex.qatools.allure.annotations.Features;
+import ru.yandex.qatools.allure.annotations.Stories;
+
+/**
+ * Test the entity listener interceptor.
+ */
+@Features("Component Tests - Repository")
+@Stories("Entity Listener Interceptor")
+public class EntityInterceptorListenerTest extends AbstractJpaIntegrationTest {
+
+    @Test
+    @Description("Verfies that the pre persist is called after a entity creation.")
+    public void prePersistIsCalledWhenPersistingATarget() {
+        executePersistAndAssertCallbackResult(new PrePersistEntityListener());
+    }
+
+    @Test
+    @Description("Verfies that the post persist is called after a entity creation.")
+    public void postPersistIsCalledWhenPersistingATarget() {
+        executePersistAndAssertCallbackResult(new PostPersistEntityListener());
+    }
+
+    @Test
+    @Description("Verfies that the post load is called after a entity is loaded.")
+    public void postLoadIsCalledWhenLoadgATarget() {
+        final PostLoadEntityListener postLoadEntityListener = new PostLoadEntityListener();
+        EntityInterceptorHolder.getInstance().getEntityInterceptors().add(postLoadEntityListener);
+
+        final Target targetToBeCreated = entityFactory.generateTarget("targetToBeCreated");
+
+        targetManagement.createTarget(targetToBeCreated);
+
+        final Target loadedTarget = targetManagement.findTargetByControllerID(targetToBeCreated.getControllerId());
+        EntityInterceptorHolder.getInstance().getEntityInterceptors().remove(postLoadEntityListener);
+        assertThat(postLoadEntityListener.getEntity()).isNotNull();
+        assertThat(postLoadEntityListener.getEntity()).isEqualTo(loadedTarget);
+    }
+
+    @Test
+    @Description("Verfies that the pre update is called after a entity update.")
+    public void preUpdateIsCalledWhenUpdateATarget() {
+        executeUpdateAndAssertCallbackResult(new PreUpdateEntityListener());
+    }
+
+    @Test
+    @Description("Verfies that the post update is called after a entity update.")
+    public void postUpdateIsCalledWhenUpdateATarget() {
+        executeUpdateAndAssertCallbackResult(new PostUpdateEntityListener());
+    }
+
+    @Test
+    @Description("Verfies that the pre remove is called after a entity deletion.")
+    public void preRemoveIsCalledWhenDeletingATarget() {
+        executeDeleteAndAssertCallbackResult(new PreRemoveEntityListener());
+    }
+
+    @Test
+    @Description("Verfies that the post remove is called after a entity deletion.")
+    public void postRemoveIsCalledWhenDeletinggATarget() {
+        executeDeleteAndAssertCallbackResult(new PostRemoveEntityListener());
+    }
+
+    private void executePersistAndAssertCallbackResult(final AbstractEntityListener entityInterceptor) {
+        final Target targetToBeCreated = entityFactory.generateTarget("targetToBeCreated");
```

Comment:
Why you're creating a target here? `addListenerAndCreateTarget(entityInterceptor, targetToBeCreated);` is doing this as well. It even returns it.